sequence_processing_pipeline/Job.py

```diff
@@ -355,3 +357,42 @@ def _group_commands(self, cmds):
             results.append(chained_cmd)
 
         return results
+
+    # assume for now that a corresponding zip file exists for each html
+    # file found. Assume for now that all html files will be found in a
+    # 'filtered_sequences' or 'trimmed_sequences' subdirectory.
+    #
+    # verify that the entire list of sample-ids found match what's
+    # expected. Since the list of expected ids is very small, we'll
+    # perform an exact comparison.
+
+    def audit(self, sample_ids):
+        """
+        Audit the results of a run.
+        :param sample_ids: A list of sample-ids that require results.
+        :return: A list of sample-ids that were not found.
+        """
+        files_found = []
+
+        if self.suffix is None:
+            raise PipelineError("Audit() method called on base Job object.")
+
+        for root, dirs, files in walk(self.output_path):
+            files_found += [join(root, x) for x in files if
+                            x.endswith(self.suffix)]
+
+        # remove all files found with a 'zero_files' directory in the path
+        files_found = [x for x in files_found if 'zero_files' not in x]
```

Comment:
What about this?
```suggestion
        for root, dirs, files in walk(self.output_path):
            if root == 'zero_files':
                continue
            files_found += [join(root, x) for x in files if
                            x.endswith(self.suffix)]
```

it should prevent hitting the issue that a sample has zero_files as part of their name